ops/pebble.py

```diff
@@ -2017,6 +2076,12 @@ def push(
             group_id: Group ID (GID) for file.
             group: Group name for file. Group's GID must match group_id if
                 both are specified.
+
+        Raises:
+            PathError: If there was an error writing the file to the path, for example, if the
```

Comment:
I think these `PathError` annotations are probably good here, though, as they're very specific to these file operations.